app/uk/gov/hmrc/thirdpartyapplication/controllers/ApplicationController.scala

```diff
@@ -157,7 +157,7 @@ class ApplicationController @Inject()(val applicationService: ApplicationService
   def deleteClientSecrets(appId: java.util.UUID) = {
     Action.async(BodyParsers.parse.json) { implicit request =>
       withJsonBody[DeleteClientSecretsRequest] { secretsRequest =>
-        credentialService.deleteClientSecrets(appId, secretsRequest.secrets).map(_ => NoContent) recover recovery
+        credentialService.deleteClientSecrets(appId, secretsRequest.actorEmailAddress, secretsRequest.secrets).map(_ => NoContent) recover recovery
```

Comment:
Can we get the identity of the requesting user from the token rather than including an email address in the body of the message? Server shouldn't trust its clients and all that...

Reply:
TPA only has a token if the request is coming from gatekeeper. Adding and deleting client secrets is meant to be used from devhub. It would probably be great if we had a token we could use for dev hub as well, but that's way out of the scope of this ticket

Reply:
Fair enough. Do we need to add something to the backlog to fix that then?

Reply:
Yeah, it may be a bit tricky as gatekeeper uses auth and devhub uses TPD for authorisation, but we need to look into this at some point. It would be great if we had something in the backlog.